### PR TITLE
Add test for rdivmod on EA array (GH23287)

### LINKED ISSUE #23287: TST: rdivmod on EA Arrays
xref #23271 

I think we need to add a test case for rdivmod as its now defined on Series.

### PR DESCRIPTION
**Pull request by makbigc**
-  closes #23287 
-  test added
-  passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/extension/decimal/test_decimal.py [hunk 1]
```diff
@@ -23,6 +23,11 @@ def data():
     return DecimalArray(make_data())
 
 
+@pytest.fixture
```
[the remark this answers is not in the document]
Comment (TomAugspurger) [reply pandas/tests/extension/decimal/test_decimal.py]: I think either of those makes sense.

Would the default be to `raise NotImpelmentedError` like the other `data` fixtures?

[the remark this answers is not in the document]
Comment (makbigc) [reply pandas/tests/extension/decimal/test_decimal.py]: Then I don't fixturize and move each array of ones inside every test functions?

[the remark this answers is not in the document]
Comment (jreback) [reply pandas/tests/extension/decimal/test_decimal.py]: @makbigc no, fixturize, but just rename it, alternatively have a fixture for each EA that returns a scalar, then inside the test functions you can create a list of those.

**Comment by jreback** [comment]
can you merge master and rebase

File: pandas/tests/extension/decimal/test_decimal.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jreback) [reply pandas/tests/extension/decimal/test_decimal.py]: can you rename

File: pandas/tests/extension/test_period.py [hunk 2]
```diff
@@ -125,6 +125,9 @@ def test_direct_arith_with_series_returns_not_implemented(self, data):
         result = data.__sub__(other)
         assert result is NotImplemented
 
+    def test_divmod_series_array(self):
```
Comment (jreback) [inline pandas/tests/extension/test_period.py]: this should work, no?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by WillAyd** [comment]
@makbigc can you merge master on this one?

File: pandas/tests/extension/test_categorical.py [hunk 3]
```diff
@@ -210,6 +210,9 @@ def test_add_series_with_extension_array(self, data):
         with pytest.raises(TypeError, match="cannot perform"):
             ser + data
 
+    def test_divmod_series_array(self):
+        pass
```
Comment (jreback) [inline pandas/tests/extension/test_categorical.py]: can you add a comment (skipping because its NotImplemented)

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
@jbrockmendel can you have a look

**Comment by jbrockmendel** [comment]
Looks fine to me.  I'm not particularly well-versed in this directory of the tests.

**Review by TomAugspurger — COMMENTED** [review]
I think you should add a fixture to `pandas/tests/extensions/conftest.py`. The docstring should describe the expected semantics of the `data_for_twos` fixture.

**Comment by jreback** [comment]
this looks fine, can you merge master and get tests to pass.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @makbigc

**Pull request merged**